crates/containerd-shim-wasm/src/sandbox/shim/otel.rs

```diff
@@ -132,33 +154,217 @@ impl Drop for ShutdownGuard {
     }
 }
 
-#[derive(Default)]
-pub struct ConfigBuilder {
-    otel_endpoint: Option<String>,
-    otel_protocol: Option<String>,
+/// Sets the OTLP endpoint from environment variables.
+fn traces_endpoint_from_env() -> anyhow::Result<String> {
+    Ok(env::var(OTEL_EXPORTER_OTLP_TRACES_ENDPOINT)
+        .or_else(|_| env::var(OTEL_EXPORTER_OTLP_ENDPOINT))?)
+}
+
+/// Sets the OTLP protocol from environment variables.
+fn traces_protocol_from_env() -> anyhow::Result<Protocol> {
+    let traces_protocol = env::var(OTEL_EXPORTER_OTLP_TRACES_PROTOCOL).unwrap_or(
+        env::var(OTEL_EXPORTER_OTLP_PROTOCOL)
+            .unwrap_or(OTEL_EXPORTER_OTLP_PROTOCOL_DEFAULT.to_owned()),
+    );
+    let protocol = match traces_protocol.as_str() {
+        OTEL_EXPORTER_OTLP_PROTOCOL_HTTP_PROTOBUF => Protocol::HttpBinary,
+        OTEL_EXPORTER_OTLP_PROTOCOL_GRPC => Protocol::Grpc,
+        _ => Err(TraceError::from(
+            "Invalid OTEL_EXPORTER_OTLP_PROTOCOL value",
+        ))?,
```

Comment:
**question:** We never create the `HttpJson` variant?

Reply:
added